plugin/storage/es/spanstore/reader.go

```diff
@@ -135,24 +136,51 @@ type timeRangeIndexFn func(indexName string, startTime time.Time, endTime time.T
 
 type sourceFn func(query elastic.Query, nextTime uint64) *elastic.SearchSource
 
-func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
+func getTimeRangeIndexFn(archive, useReadWriteAliases bool, remoteReadClusters []string) timeRangeIndexFn {
 	if archive {
 		var archivePrefix string
 		if useReadWriteAliases {
 			archivePrefix = archiveReadIndexSuffix
 		} else {
 			archivePrefix = archiveIndexSuffix
 		}
-		return func(indexName string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indexName string, startTime time.Time, endTime time.Time) []string {
 			return []string{archiveIndex(indexName, archivePrefix)}
-		}
+		}, remoteReadClusters)
 	}
+
 	if useReadWriteAliases {
-		return func(indices string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indices string, startTime time.Time, endTime time.Time) []string {
 			return []string{indices + "read"}
+		}, remoteReadClusters)
+	}
+
+	return addRemoteReadClusters(timeRangeIndices, remoteReadClusters)
+}
+
+// Elasticsearch cross cluster query api example: GET /twitter,cluster_one:twitter,cluster_two:twitter/_search
+// Add a remote cluster prefix for each cluster and for each index.
+// The special remote cluster "local" leaves the prefix off of it's subset of indexes resulting in a standard local query
+func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) timeRangeIndexFn {
+	return func(indices string, startTime time.Time, endTime time.Time) []string {
+		jaegerIndices := fn(indices, startTime, endTime)
+		if len(remoteReadClusters) > 0 {
+			remoteJaegerIndices := []string{}
```

Comment:
We know the size at creation time

Reply:
Do you want the array to be statically sized?
remoteJaegerIndices := [size]string{}

Reply:
yes

Reply:
After removing local just appended to the jaegerIndices to simplify things.